GiniVision/Classes/CameraViewController.swift

```diff
@@ -303,60 +316,86 @@ public typealias CameraErrorBlock = (_ error: CameraError) -> ()
         }
         self.previewView.addSubview(imageView)
         UIView.animate(withDuration: 1.5,
-                                   animations: {
-                                    imageView.alpha = 0.0
-            },
-                                   completion: { (success: Bool) -> Void in
-                                    imageView.removeFromSuperview()
+                       animations: {
+                        imageView.alpha = 0.0
+        },
+                       completion: { (success: Bool) -> Void in
+                        imageView.removeFromSuperview()
         })
     }
     
     // MARK: Constraints
     fileprivate func addConstraints() {
-        let superview = self.view
-        
-        // Preview view
+        addPreviewViewConstraints()
+        addCameraOverlayConstraints()
+        addControlsViewConstraints()
+        addCaptureButtonConstraints()
+    }
+    
+    fileprivate func addPreviewViewConstraints(){
         previewView.translatesAutoresizingMaskIntoConstraints = false
         
-        // lower priority constraints - will make the preview "want" to get bigger
-        ConstraintUtils.addActiveConstraint(item: previewView, attribute: .top, relatedBy: .equal, toItem: superview, attribute: .top, multiplier: 1, constant: 0, priority: 1000)
-        ConstraintUtils.addActiveConstraint(item: previewView, attribute: .bottom, relatedBy: .equal, toItem: superview, attribute: .bottom, multiplier: 1, constant: 0, priority: 750)
-        ConstraintUtils.addActiveConstraint(item: previewView, attribute: .left, relatedBy: .equal, toItem: superview, attribute: .left, multiplier: 1, constant: 0, priority: 750)
-        ConstraintUtils.addActiveConstraint(item: previewView, attribute: .right, relatedBy: .equal, toItem: superview, attribute: .right, multiplier: 1, constant: 0, priority: 750)
-        
-        // required constraints - make sure the preview doesn't expand into other views or off-screen
-        ConstraintUtils.addActiveConstraint(item: superview!, attribute: .bottom, relatedBy: .greaterThanOrEqual, toItem: previewView, attribute: .bottom, multiplier: 1, constant: 0, priority: 1000)
-        ConstraintUtils.addActiveConstraint(item: previewView, attribute: .left, relatedBy: .greaterThanOrEqual, toItem: superview, attribute: .left, multiplier: 1, constant: 0, priority: 1000)
-        ConstraintUtils.addActiveConstraint(item: superview!, attribute: .right, relatedBy: .greaterThanOrEqual, toItem: previewView, attribute: .right, multiplier: 1, constant: 0, priority: 1000)
-        
-        ConstraintUtils.addActiveConstraint(item: previewView, attribute: .width, relatedBy: .equal, toItem: previewView, attribute: .height, multiplier: 3/4, constant: 0)
-        ConstraintUtils.addActiveConstraint(item: previewView, attribute: .centerX, relatedBy: .equal, toItem: superview, attribute: .centerX, multiplier: 1, constant: 0)
+        if UIDevice.current.userInterfaceIdiom == .pad{
```

Comment:
This check for tablet or phone is a bit ugly. I'm not sure we can do it in any other way, but there might be. Can you try to think of an alternative?